model/project_parser.go

```diff
@@ -898,6 +900,24 @@ func retrieveFileForModule(ctx context.Context, opts GetProjectOpts, modules Mod
 		ReadFileFrom: ReadFromGithub,
 		Identifier:   include.Module,
 	}
+
+	// If provided a patch to repeat, use the same githash as the original patch did for the module.
```

Comment:
Is there a world where the patch that's being repeated would have included changes, such that we need to make sure we're applying those changes on top as well? (Or is that handled elsewhere?)

Reply:
We'd like that to work. I think DEVPROD-9824 is maybe the ticket for that, although it looks like DEVPROD-9870 and DEVPROD-11466 may be duplicates if I'm understanding this change correctly? Like does this also fix DEVPROD-11466?

Reply:
@ablack12 I may be misunderstanding the code but it kind of looks like we don't preserve the LocalModuleIncludes from the initial call to patch, so there's not an easy way to reconstruct the patched config when we repeat. Is that right?

Reply:
It seems like this is easily solvable by just changing the value [here](https://github.com/evergreen-ci/evergreen/blob/main/model/patch/patch.go#L195) to be saved in the DB? Otherwise you'd have to reconstruct the module files from the patch data and that seems stressful, and is what was avoided when @bynn initially added the local functionality.

Reply:
this change would only fix it for the case where the module set is committed and can be pulled from github 

@johndaniels is right in that we do not save the local module changes anywhere and wouldn't make sense to save every local module change just in case a user would want to repeat the patch. I think for the purpose of this ticket it should be okay for it to only work for committed changes.

I also don't think DEVPROD-9870 and DEVPROD-11466 are duplicates since the problems are a bit different (but zack is working on it right now")

Reply:
Ah I see. They are different, although it seems like very related in the code.

Is there a path forward on [DEVPROD-9824](https://jira.mongodb.org/browse/DEVPROD-9824). Am I understanding correctly that that's the ticket for duplicating module configs that are only locally patched?

Reply:
For DEVPROD-9824 the solution you mentioned above works but I'd want to backlog it unless the workaround of committing the module changes for cloning is not enough for the workflow.

Reply:
Hey, just to +1 what John mentioned, it's important that we get the exact same patch when we're using multi-patch. This way we can run repeated perf runs to get the same results without having to take additional steps in our workflow.

Reply:
it makes sense to do the ticket if it's important for the workflow! could I ask you guys to comment on the relevant ticket so we know how to prioritize them?